Adding samples

stuarthendren opened the pull request:
Based on examples in the ies4 document.

I was running through these as an exercise to learn the modelling and thought it would make sense to add them to the repo.
There are a few differences, happy to change as preferred:

- Generally use `_` in IRIs rather that %20 as easier to read. 
- The existing example uses `http://iso.org/iso8601#`, so I've done the same, but the example in the docs uses `http://iso8601.iso.org#`. 
- Changed the date in the sometimes example so the end is after the start.
- Used blank nodes in Characteristics and Measure example.
- I haven't done the Metadata example, wasn't sure what the `dc:` elements would be.
- Movement example - unsure of the extended data period.
- Corrects `rdfs:subClassOf` reference in Types example.
- I've not use any particular formatter.

stuarthendren committed:
Adding samples

Based on examples in the ies4 document.

stuarthendren committed:
Adds missing types from identifier example

stuarthendren committed:
Fix reference to isStateOf in preiod-of-time sample

stuarthendren commented:
I also noticed that the relationships and period-of-time examples use ies:Employed but that doesn't appear to be in the ontology?

stuarthendren committed:
Add Organisation type to relationship example

stuarthendren committed:
Fix incorrect stateOf to isStateOf in relationship example

stuarthendren committed:
Fix iso8601 prefix reference in types example

stuarthendren committed:
Fix spelling error in assessment example

stuarthendren committed:
Spelling and class fixes

Changes schemeImplementBy to schemeMasteredIn as implemented by seems to have been replaced

MonkeyChap commented:
Hi - the %20 encoding is where there are spaces in the existing standard the URI is based on. I think for 8601 we need to do much more about mandating the specific format - e.g. use of Z, T instead of space, etc.

paulc-dstl commented:
> Hi - the %20 encoding is where there are spaces in the existing standard the URI is based on. I think for 8601 we need to do much more about mandating the specific format - e.g. use of Z, T instead of space, etc.

@MonkeyChap agreed re syntax of 8601 strings. We must have a syntax which is safe by default and does not require and additional encoding. In this specific example, it's not just spaces in the URIs, it's the colons which necessarily get encoded for use as URI components (colon is a reserved character as per [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-2.2)). 

Looking at the [8601 spec on wikipedia](https://en.wikipedia.org/wiki/ISO_8601), there are options which would be natively web safe ie produce URIs which do not require further encoding. And this would apply to both positions in time as well as durations.

paulc-dstl commented:
@stuarthendren apologies for taking so long to get onto this. Thanks for your input. Am getting on with a review.

paulc-dstl reviewed (dismissed):
These additions make the type assertions explicit.

paulc-dstl commented:
> These additions make the type assertions explicit.

Hmm. I reviewed the type assignments commit, but the review applies to all commits in the PR. So i have withdrawn the review and will look at each commit in turn before submitting the review again. Sorry about that. 

I've also added @afizzycola for a second review as some of the additions look to be compliant IES but not sure if they conform to the established patterns (for example the role of chair in a meeting).

paulc-dstl reviewed (commented):
Detailed comments on commits to follow:

paulc-dstl commented on Sample Data/identifiers.ttl:
This looks good. Makes the data assignments explicit. Of course, if the domain and ranges are set appropriately, these types could be inferred (if the range of ies:isIdentifiedBy is ies:Identifier for example).

So I'm happy with this commit.

paulc-dstl commented on Sample Data/period-of-time.ttl:
Simple error fix. Approve.

paulc-dstl commented on Sample Data/relationships.ttl:
Another explicit type assignment. See previous comment on such additions.

afizzycola commented on Sample Data/assessment.ttl:
```diff
@@ -0,0 +1,56 @@
+  a ies:Perpetrator .
+
+data:S1Assessment ies:assessed data:Scenario1 ;
+  ies:confidence "HIGH" ;
```
The confidence predicates here are ```ies:hmlConfidence``` rather than the more generic ```ies:confidence``` `` predicate. Admittedly, the diagram in the document doesn't help as it does not specify them - I have made a not of this and will update

afizzycola replied:
As promised, I have now made the change to the diagram in the standard

afizzycola commented on Sample Data/period-of-time.ttl:
```diff
@@ -0,0 +1,17 @@
+@prefix iso8601: <http://iso.org/iso8601#> .
+
+data:Fred a ies:Person .
+data:FredState2 a ies:Employed ;
```
Oops there is a mistake in the example in the standards document. There is no ```ies:Employed``` in IES4. If you could change this to a ```ies:PersonState``` I will make the change in the docs for IES 4.3

afizzycola replied:
I have made this change for IES 4.3

afizzycola commented on Sample Data/period-of-time.ttl:
```diff
@@ -0,0 +1,17 @@
+data:Fred a ies:Person .
+data:FredState2 a ies:Employed ;
+    ies:isStateOf data:Fred ;
+    ies:worksFor data:AcmeInc .
```
```data:AcmeInc``` is not instantiated as a ```ies:Organisation```

afizzycola commented on Sample Data/relationships.ttl:
```diff
@@ -0,0 +1,16 @@
+data:Barry a ies:Person .
+data:Fred a ies:Person ;
+    ies:siblingOf data:Barry .
+data:FredState2 a ies:Employed ;
```
Same as the above comment, could we make this ```ies:PersonState```?

afizzycola commented on Sample Data/sometimes.ttl:
The datetime URI just need instantiating here. I will reflect the same change you made to the end time

afizzycola commented on Sample Data/sometimes.ttl:
```diff
@@ -0,0 +1,16 @@
+  ies:isStartOf data:LK176LPX_in_Acacia_Gardens ;
+  ies:inPeriod iso8601:1987-06-01T09%3A00 .
+data:LK176LPX_stops_parking a ies:BoundingState ;
+  ies:isStartOf data:LK176LPX_in_Acacia_Gardens ;
```
should be ```ies:isEndOf```

afizzycola reviewed (changes requested):
@stuarthendren thank you so much for creating these, they will be really handy for first time users of IES. Your work has also revealed some inconsistencies in the documentation which I will address in IES 4.3. Also, my apologies for taking so long to review this

paulc-dstl commented:
@stuarthendren @afizzycola - just checking on the status of this - is everything now resolved, changes made, etc? TIA :-)

stuarthendren commented:
Hi Paul - I had actually missed that there had been a review from @afizzycola - I can fix up those things, will take a look tomorrow. Though I'm not sure if other things have moved on now with the other recent changes - but I'll try to take account of those too.

stuarthendren committed:
Fixing PR comments

stuarthendren committed:
Updates temporal references

Now use the unpunctuated form of the temporal references are recommended for 4.3.

afizzycola reviewed (changes requested):
All looks good apart from one class of thing. 4.3 now mandates that you do not put a trailing Z in the ParticularPeriod URIs as it's assumed that all time is in UTC. Also Zs can't be used in for partial ISO8601 strings where you don't articulate a time e.g ```2025-12-13```

rebecca-dstl reviewed (dismissed):
Awaiting requested changes requested

paulc-dstl commented:
Hi @stuarthendren - I think this is just waiting on a couple of changes requested by @afizzycola and @rebecca-dstl - if these could be actioned and the PR reviewed and merged, that would be most helpful (I am looking to do a minor 3.4.1 release ASAP with all outstanding actions complete). 

Many thanks :-)

stuarthendren commented:
Hi @paulc-dstl, seems I'm only getting notifications if mentioned so missed the other comments again. 

Yeah, I can make those changes

stuarthendren commented:
@afizzycola , @paulc-dstl  You suggest removing the Z from the ParticularPeriod but wouldn't that be ambiguous? and it is included int e example from the ies.md - 

> 10. Changes to Period of Time pattern:
ParticularPeriod URI pattern now mandated to be non-punctuated encoding (20070118T153000Z). This avoids the use of escape characters in the URI. NOTE: the literal for iso8601PeriodRepresentation remains punctuated.

stuarthendren commented:
I see the point below the quote above makes it mandatory to be in Z

> ParticularPeriod mandated to be in UTC / Zulu time

So maybe it's not ambiguous but the example should be updated instead?

stuarthendren committed:
Updating ParticularPeriod instance inline with v4.3.0

The examples have dropped the Z and removed the punctuation
to conform to the recommended approach

stuarthendren commented:
I haven't updated the ies.md as discussed above as the version I have is out of date and feel a bit out of my purview.

paulc-dstl reviewed (approved):
Looks good, thanks.

afizzycola reviewed (approved):
Thanks for the updates @stuarthendren. All review points now rectified
